pi-blaster.c

```diff
@@ -70,23 +72,32 @@ static uint8_t pin2gpio[8];
 // will use too much memory bandwidth.  10us is a good value, though you
 // might be ok setting it as low as 2us.
 
-#define CYCLE_TIME_US		10000
+#define CYCLE_TIME_US	10000
 #define SAMPLE_US		10
 #define NUM_SAMPLES		(CYCLE_TIME_US/SAMPLE_US)
 #define NUM_CBS			(NUM_SAMPLES*2)
 
-#define NUM_PAGES		((NUM_CBS * 32 + NUM_SAMPLES * 4 + \
+#define NUM_PAGES		((NUM_CBS * sizeof(dma_cb_t) + NUM_SAMPLES * 4 + \
 					PAGE_SIZE - 1) >> PAGE_SHIFT)
 
-#define DMA_BASE		0x20007000
-#define DMA_LEN			0x24
-#define PWM_BASE		0x2020C000
+#define DMA_BASE		(periph_virt_base + 0x00007000)
+#define DMA_CHAN_SIZE	0x100 /* size of register space for a single DMA channel */
+#define DMA_CHAN_MAX	14  /* number of DMA Channels we have... actually, there are 15... but channel fifteen is mapped at a different DMA_BASE, so we leave that one alone */
+#define DMA_CHAN_NUM	14  /* the DMA Channel we are using, NOTE: DMA Ch 0 seems to be used by X... better not use it ;) */
```

Comment:
I [posted the question on StackOverflow](http://stackoverflow.com/questions/29628602/how-to-allocate-dma-channel-in-user-space). We will see if we can come up with a good definitive answer.

Also found (afterwards) [another interesting answer](http://stackoverflow.com/questions/28868932/using-dma-api-in-linux-kernel-but-channel-is-never-available) which mentions looking in `/sys/class/dma/<channel>/in_use`. I do not have a Pi handy right now. Give it a try if you can and still have time, otherwise I am fine with hardcoding chan 14.